build.gradle

```diff
@@ -97,16 +97,58 @@ subprojects {
 
   apply plugin: "com.diffplug.gradle.spotless"
   spotless {
-    lineEndings = 'unix';
+    lineEndings = 'unix'
     java {
       target project.fileTree(project.projectDir) {
         include '**/*.java'
         exclude '**/generated-src/**'
       }
+
+      custom 'Remove commented-out import statements', {
+        it.replaceAll(/\n\/\/ import .*?;.*/, '')
+      }
+
+      custom 'End files with a single blank line', {
+        it.replaceAll(/\s+$/, '\n')
+      }
+
+      // Removes end-of-line whitespace
+      trimTrailingWhitespace()
+
+      // Enforce style import order
+      importOrderFile "${project(':geode-core').projectDir}/../etc/eclipseOrganizeImports.importorder"
+
+      // Enforce style modifier order
+      custom 'Modifier ordering', {
+        def modifierRanking = [
+                public      : 1,
+                protected   : 2,
+                private     : 3,
+                abstract    : 4,
+                default     : 5,
+                static      : 6,
+                final       : 7,
+                transient   : 8,
+                volatile    : 9,
+                synchronized: 10,
+                native      : 11,
+                strictf     : 12]
+        // Find any instance of multiple modifiers. Lead with a non-word character to avoid
+        // accidental matching against for instance, "an alternative default value"
+        it.replaceAll(/\W(?:public |protected |private |abstract |default |static |final |transient |volatile |synchronized |native |strictfp ){2,}/, {
```

Comment:
I'm confused by the comment. Is this matching the whole line? The way I'm reading it, it will match "default" in "an alternative default value", but replace it with "default". Will this match comments?

Reply:
Line 318 is attempting to match any substring that (1) starts with a non-word character and (2) contains two or more modifiers.  That substring then has the match modifiers reordered.

The non-word character is required to avoid finding `native default` in the string "alternative default value` and "reordering" it to "alterdefault native value".

This will match comments, and indeed the "an alternative default value" was found in a docstring in `BaseValueHolder` test.  At time of development, the only comments matched were commented-out code with incorrect modifiers.

I suppose it could be extended to also include optional whitespace after a carriage return to avoid picking up comments or javadocs.  Would this be preferable?

Reply:
If I wanted to be contrarian, I could write` /* HAHA! */static public void main`.

Reply:
Now that you mention it, that contrarian scenario would get a space added between the `*/` and the `static` by the formatter, and then get reordered on a subsequent spotless, which would mean `spotlessApply` followed by `spotlessCheck` could theoretically fail.

Maybe contrarians deserve that failure, but I should promote the eclipseFormatter to before the modifier reordering.